pac4j-core/src/main/java/org/pac4j/core/client/Client.java

```diff
@@ -56,7 +58,14 @@
      * @param context web context
      * @return the user profile
      */
-    U getUserProfile(C credentials, WebContext context);
+    Optional<U> getUserProfile(C credentials, WebContext context);
+
+    default Optional<U> getUserProfile(Optional<C> credentials, WebContext context) {
+        if(credentials.isPresent()) {
```

Comment:
use map()

Reply:
I'm questioning which `getUserProfile` should be default or removed: with optional or accepting null?

Reply:
Why do we have two `getUserProfile` methods in the `Client` interface? This is the most important class of *pac4j*: we should keep it as clean as possible.